### PR TITLE
Update for SwiftSyntax regex changes

### PR DESCRIPTION
**Pull request by hamishknight**
Update for https://github.com/apple/swift-syntax/pull/1463

### CODE REVIEW

**Comment by hamishknight** [comment]
https://github.com/apple/swift-syntax/pull/1463

@swift-ci please test

File: Sources/SourceKitLSP/DocumentTokens.swift [hunk 1]
```diff
@@ -96,6 +96,8 @@ extension SyntaxClassification {
       return (.number, [])
     case .stringLiteral:
       return (.string, [])
+    case .regexLiteral:
+      return (.regexp, [])
```
[the remark this answers is not in the document]
Comment (hamishknight) [reply Sources/SourceKitLSP/DocumentTokens.swift:100]: Hmm, seems the case was added before regex literals were implemented...

[the remark this answers is not in the document]
Comment (hamishknight) [reply Sources/SourceKitLSP/DocumentTokens.swift:100]: Okay best I can tell the answer to "I don't really know how highlighting for regexes would have worked before" is that we didn't have any special handling, so nice that it's now handled.

**Comment by hamishknight** [comment]
https://github.com/apple/swift-syntax/pull/1463

@swift-ci please test

**Comment by hamishknight** [comment]
https://github.com/apple/swift-syntax/pull/1463

@swift-ci please test Windows

File: Sources/SourceKitLSP/DocumentTokens.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (ahoppen) [reply Sources/SourceKitLSP/DocumentTokens.swift:100]: Yes, looks correct to me

**Review by ahoppen — APPROVED** [review]

**Review by MaxDesiatov — APPROVED** [review]
👍

**Pull request merged**